.github/workflows/tests/BUILD

```diff
@@ -0,0 +1,24 @@
+# COMMENT
+
+shell_sources(name="tests")
+
+def _run_test(filename):
+    filepath = build_file_dir() / 'smoke_test_auth_cherry_picker.sh'
+    return f"chmod +x {filepath} && {filepath}"
+
+
+
+experimental_test_shell_command(
+    name="Test auto-cherry-picker prereqs",
+    command=_run_test("smoke_test_auth_cherry_picker.sh"),
+    tools=["chmod", "cat"],
+    execution_dependencies=[
+        "//build-support/bin/act:act-at-root",
+        "./smoke_test_auth_cherry_picker.sh",
+        ".github/workflows/auto-cherry-picker.yaml",
+        "//build-support/bin/cherry_pick/package.json:support_files",
+        "//build-support/bin/cherry_pick/package-lock.json:support_files",
```

Comment:
Idea for discussion: arguably these dependencies are in the wrong place: 

- the `smoke_test_auth_cherry_picker.sh` script should depend on act and the workflow
- the workflow itself should depend on `//build-support/bin/cherry_pick/package.json:support_files` (and the package-lock.json), if executing the workflow requires them (or maybe they're not needed at all, for this test?)

If they're moved, then this can be just `execution_dependencies=["./smoke_test_auth_cherry_picker.sh"]`... but, this might all be made redundant by moving to `shunit2_tests` anyway.

Reply:
I had that originally, but you get a much wider cache key that way.

I decided I was probably worth breaking the modeling to get better cache rates.

Thoughts?

Reply:
Could you expand on your concern? 

I would've thought the cache behaviour ends up the same (in terms of how often the test is invalidated/has to rerun), assuming that the dependencies are only attached to the specific workflows that need them: that is, `overrides` to the generator, rather than top-level `dependencies`. In that case, the files transitively depended upon by the test are the same. I may be misunderstanding.

Reply:
Oh I misread your original post. Yeah that's better 😅